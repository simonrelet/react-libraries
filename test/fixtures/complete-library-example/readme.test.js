'use strict'

const fs = require('fs-extra')
const path = require('path')
const { Fixture } = require('../../fixtureUtils')

describe(`${path.basename(__dirname)} readme`, () => {
  let fixture

  beforeAll(async () => {
    fixture = new Fixture(__dirname)
    await fixture.initialize()
  })

  afterAll(async () => {
    await fixture.reset()
  })

  it('should not fail', async () => {
    expect(await fixture.runScript('readme')).toMatchSnapshot()
  })

  it('should generate README', async () => {
    expect(await fs.readFile(fixture.paths.readme, 'utf8')).toMatchSnapshot()
  })
})
