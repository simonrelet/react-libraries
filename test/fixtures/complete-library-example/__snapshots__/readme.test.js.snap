// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complete-library-example readme should generate README 1`] = `
"<!--
  THIS FILE WAS GENERATED!
  Don't make any changes in it, update README-template.md instead.
  If you still need to make changes in this file, remove this header so it won't be overridden.
-->

# complete-library-example

This is an example of library.

## Installation

### npm

\`\`\`sh
npm install --save complete-library-example
\`\`\`

### yarn

\`\`\`sh
yarn add complete-library-example
\`\`\`

## Documentation

The documentation can be found [here](https://github.com/complete-library-example/complete-library-example/tree/1.0.0/docs).
"
`;

exports[`complete-library-example readme should not fail 1`] = `
Exit code: 0

--- Output ---
README-template.md -> README.md
--------------
`;
